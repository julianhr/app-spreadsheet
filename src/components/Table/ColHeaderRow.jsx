/* @jsx jsx */
import React from 'react'
import PropTypes from 'prop-types'
import styled from '@emotion/styled'
import { jsx, css } from '@emotion/core'

import ColHeader from '../cell/ColHeader'
import { ROW_HEADER_WIDTH } from '../cell/RowHeader'


const Row = styled.div`
  display: flex;
  width: fit-content;
  position: sticky;
  top: 0;
`

const InnerBorder = styled.div`
  z-index: 20;
  display: flex;
  align-items: center;
  justify-content: center;
  border-right: 1px solid ${props => props.theme.colors.table.borderDark};
  border-bottom: 1px solid ${props => props.theme.colors.table.borderDark};
  background: ${props => props.theme.colors.table.labelBkg};
`

function ColHeaderRow({ colLabels }) {
  const renderCornerCell = () => (
    <InnerBorder
      css={css`
        width: ${ROW_HEADER_WIDTH}px;
        height: 26px;
      `}
    />
  )

  const renderLabelCells = () => (
    new Array(colLabels.length).fill(0).map((_, i) => (
      <ColHeader
        key={i}
        label={colLabels[i]}
      />
    )
  ))

  return (
    <Row
      data-row='header'
    >
      {renderCornerCell()}
      {renderLabelCells()}
    </Row>
  )
}

ColHeaderRow.propTypes = {
  colLabels: PropTypes.arrayOf(PropTypes.string)
}

export default ColHeaderRow

