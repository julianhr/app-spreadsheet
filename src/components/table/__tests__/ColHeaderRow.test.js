import React from 'react'
import { render, fireEvent, cleanup } from '@testing-library/react'

import MockApp from '~/__tests__/__mocks__/MockApp'
import ColHeaderRow from '../ColHeaderRow'


const renderApp = (props) => {
  const wrapper = render(
    <MockApp>
      <table>
        <ColHeaderRow {...props} />
      </table>
    </MockApp>
  )

  return wrapper
}

describe('ColHeaderRow', () => {
  afterEach(cleanup)

  it('matches snapshot', () => {
    const colLabels = ['A', 'B']
    const props = { colLabels }
    renderApp(props)
    const el = document.querySelector('[data-row="header"]')
    expect(el).toMatchSnapshot()
  })
})

