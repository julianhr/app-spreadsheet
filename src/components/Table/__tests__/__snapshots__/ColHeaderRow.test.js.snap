// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColHeaderRow matches snapshot 1`] = `
<div
  class="css-wkndpb-Row eft810l0"
  data-row="header"
>
  <div
    class="css-10tzczs-InnerBorder eft810l1"
  />
  <div
    class="col-label-height css-bow3c7-Root-ColHeader e11zdtao0"
    data-col="A"
  >
    A
    <button
      class="css-rewts0-BtnWidth e11zdtao1"
      tabindex="-1"
    />
  </div>
  <div
    class="col-label-height css-bow3c7-Root-ColHeader e11zdtao0"
    data-col="B"
  >
    B
    <button
      class="css-rewts0-BtnWidth e11zdtao1"
      tabindex="-1"
    />
  </div>
</div>
`;
